src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,139 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private Object[] objects;
+    private int size;
+
+    public ArrayList() {
+        size = 0;
+        objects = new Object[10];
+    }
+
     @Override
     public void add(T value) {
+        if (size == objects.length) {
+            Object[] newObject = new Object[objects.length + (int) objects.length / 2];
+            for (int i = 0; i < objects.length; i++) {
+                newObject[i] = objects[i];
+            }
+            objects = newObject;
+        }
 
+        objects[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (size == objects.length) {
+            Object[] newObject = new Object[objects.length + (int) objects.length / 2];
+            for (int i = 0; i < objects.length; i++) {
+                newObject[i] = objects[i];
+            }
+            objects = newObject;
+        }
+        if (size == 0 && index == size) {
+            objects[0] = value;
+            size++;
+        } else if ((size > index || size == 0) && index >= 0) {
+            for (int i = size; i > index; i--) {
+                objects[i] = objects[i - 1];
+            }
+            objects[index] = value;
+            size++;
+        } else if (index == size) {
+            objects[index] = value;
+            size++;
+        } else {
+            throw new ArrayListIndexOutOfBoundsException("index invalid");
```

Comment:
The exception message 'index invalid' is not very informative. Consider providing more details, such as the invalid index value and the acceptable range.